### ISSUE TITLE
Assure port types everywhere in Nginx and Apache plugins

### ISSUE THREAD

**Issue by ohemorange** [certbot/certbot#5649]
Why are they strings in the first place? This causes bugs like #5550.

**Comment by eicksl** [comment]
Could you be more specific about how the string datatype is causing errors?

The reason for using strings seems to be because it's easier to parse and overwrite the nginx address (the part that comes after the 'listen' directive). The `Addr` class in `obj.py` uses `fromstring` and `to_string` methods for I/O.

**Comment by ohemorange** [comment]
We have an internal `Addr` object that parses ports out of address strings. The port it saves is always an `int`, even though it's being represented as a string. But the port number that comes in from certbot is actually an `int`. We sometimes need to compare these two, and that's where bugs arise. Ideally, ports will be dealt with as `int`s anytime within certbot code, and strings only when they're read in or written out.

**Comment by edaleeta** [comment]
Hi there!
I tried taking a stab at this to get acquainted with the codebase.
I noticed the common `Addr` class that the nginx `Addr` class inherits from also saves the port as an `str`.

Was hoping to get some clarification as to whether you wanted to see the type for the port get changed on nginx's `Addr` or in `configurator.py` where port comparisons occur.

**Comment by eicksl** [comment]
Yes, it seems that `common.Addr` is where the port is actually being saved. Both the nginx and apache plugins inherit from `common.Addr` and call `super` to save the port.

**Comment by ohemorange** [comment]
You're right! Updating the title to indicate that it applies to both the Nginx and Apache plugins.
